Remotion/Security/Core/Metadata/LocalizationFileNameStrategy.cs

```diff
@@ -30,8 +30,13 @@ public string GetLocalizationFileName (string metadataFilename, CultureInfo cult
       ArgumentUtility.CheckNotNull ("culture", culture);
 
       string baseFilename = Path.GetFileNameWithoutExtension (metadataFilename);
-      // TODO RM-7870: Add notnull assertion
-      string basePath = Path.GetDirectoryName (metadataFilename)!;
+
+      string? basePath = Path.GetDirectoryName (metadataFilename);
+      if (basePath == null)
+      {
+        throw new ArgumentException ("The metadata filename must denote a path that is not a root directory.", "metadataFilename");
+      }
```

Comment:
```suggestion
        throw new ArgumentException ("The metadata filename must denote a path that is not a root directory.", "metadataFilename");
```

For single-line IFs, we can (and usually should) sckip the branches to improve readdability/compactness of the code.